### PR TITLE
LPS-153777 Poshi Automation for Bug  | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-153777

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase [hunk 1]
```diff
@@ -490,4 +490,86 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-140988. The Menu Display fragment should be visible to guest with default permissions."
+	@priority = "4"
+	test ViewMenuDisplayFromMasterTemplateAsGuest {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase:495]: ViewMenuDisplayAsGuest

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/NavBar.path [hunk 2]
```diff
@@ -12,7 +12,7 @@
 <tbody>
 <tr>
 	<td>ACTIVE_NAV_ITEM_LINK</td>
-	<td>//div[contains(@class,'navbar')]//a[contains(@class,'active') and normalize-space()='${key_navItem}'] | //nav[contains(@role,'navigation')]//li[contains(@class,'active')][normalize-space()='${key_navItem}']</td>
+	<td>//div[contains(@class,'navbar')]//a[contains(@class,'active') and normalize-space()='${key_navItem}'] | //nav[contains(@role,'navigation')]//li[contains(@class,'active')][normalize-space()='${key_navItem}'] | //li[contains(@class,'nav-item') and contains(@class,'active')]//*[normalize-space(text())='${key_navItem}']</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/NavBar.path:15]: No need

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/NavBar.path [hunk 3]
```diff
@@ -82,7 +82,7 @@
 </tr>
 <tr>
 	<td>NAV_ITEM_LINK</td>
-	<td>//div[contains(@class,'navbar')]//a[normalize-space()='${key_navItem}'] | //nav[contains(@role,'navigation')]//li[normalize-space()='${key_navItem}']</td>
+	<td>//div[contains(@class,'navbar')]//a[normalize-space()='${key_navItem}'] | //nav[contains(@role,'navigation')]//li[normalize-space()='${key_navItem}'] | //li[contains(@class,'nav-item')]//*[normalize-space(text())='Test Page Name 2']</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/NavBar.path:85]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase [hunk 4]
```diff
@@ -490,4 +490,86 @@ definition {
...
+		task ("Then the guest can see the page's name on Menu Display") {
+			for (var i : list "1,2") {
+				AssertElementPresent(
+					key_navUnderline = "Test Page Name ${i}",
+					locator1 = "NavUnderline#NAV_UNDERLINE_LINK");
+			}
+
+			AssertTextNotPresent(
+				locator1 = "Message#ERROR",
+				value1 = "An unexpected error occurred");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase:560]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase [hunk 5]
```diff
@@ -490,4 +490,86 @@ definition {
...
+			}
+		}
+
+		task ("When the guest accesses the new site") {
+			User.logoutPG();
+
+			Navigator.openSiteURL(siteName = "Test Site Name");
+		}
+
+		task ("Then the guest can see the page's name on Menu Display") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase:551]: task ("Then the guest could navigate through pages via Menu Display") {
Navigator.gotoNavTab();

			ContentPages.viewFragmentText(
				fragmentName = "heading",
				id = "element-text",
				text = "Heading Example");
}

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase [hunk 6]
```diff
@@ -490,4 +490,86 @@ definition {
...
+		task ("When the user clicks on second page of Menu Display") {
+			Navigator.gotoNavItem(navItem = "Test Page Name 2");
+		}
+
+		task ("Then the user can see the Heading fragment") {
+			ContentPages.viewFragmentText(
+				fragmentName = "heading",
+				id = "element-text",
+				text = "Heading Example");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/menudisplay/MenuDisplay.testcase:572]: Redundant

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/NavBar.path (hunk 2 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/NavBar.path:15]: I think this new path is needed, non of the existing paths related to NAV works. May you double-check?

Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/NavBar.path]: They are two different elements, shouldn't combine them in one locator ACTIVE_NAV_ITEM_LINK
image
image

**Comment by Tim-Cao** [comment]
See changes https://github.com/liferay-echo/liferay-portal/pull/9310
@ozysouza

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**